lib/farside/http.ex

```diff
@@ -0,0 +1,123 @@
+defmodule Farside.Http do
+  require Logger
+
+  @moduledoc """
+  Http
+    the http client
+  """
+
+  @headers Application.fetch_env!(:farside, :headers)
+  @queries Application.fetch_env!(:farside, :queries)
+  @recv_timeout String.to_integer(Application.fetch_env!(:farside, :recv_timeout))
+
+  def request(url) do
+    cond do
+      System.get_env("FARSIDE_TEST") ->
+        :good
+
+      true ->
+        HTTPoison.get(url, @headers)
+        |> then(&elem(&1, 1))
+        |> Map.get(:status_code)
+        |> case do
+          n when n < 400 ->
+            Logger.info("Response: [#{n}]")
+            :good
+
+          n ->
+            Logger.error("Response: [#{n}]")
+            :bad
+        end
+    end
+  end
+
+  def request(url, type) do
+    cond do
+      System.get_env("FARSIDE_TEST") ->
+        :good
+
+      true ->
+        HTTPoison.get(url, @headers, timeout: 5000, recv_timeout: @recv_timeout)
+        |> then(&elem(&1, 1))
+        |> Map.get(:status_code)
+        |> case do
+          n when n < 400 ->
+            Logger.info("Type: #{type}, Response: [#{n}], Url: #{url}")
+            :good
+
+          nil ->
+            Logger.error("Type: #{type}, Response: [408], Url: #{url}")
+            :bad
```

Comment:
I'm not sure I understand the reasoning for this -- it seems to be flagging a lot of false negatives (i.e. the instance responds fine when verifying manually, but is returning a `nil` status code apparently?)

Reply:
this generally happens when the instance doesnt respond in the timeout interval, this can be adjusted in the config file